controller/execute_test.go

```diff
@@ -225,12 +225,15 @@ func TestServeMetrics(t *testing.T) {
 	_, port, _ := net.SplitHostPort(l.Addr().String())
 
 	go serveMetrics(fmt.Sprintf(":%s", port))
+	// Required to let serveMetrics go routine to open the network socket
+	time.Sleep(10 * time.Millisecond)
+
 	resp, err := http.Get(fmt.Sprintf("http://localhost:%s", port) + "/healthz")
-	assert.NoError(t, err)
+	require.NoError(t, err)
```

Comment:
mixing assert and require in same test, is there is an advantage using require?

Reply:
It avoids panic in the test.

Reply:
Panic is expected. I just did lift and shift when moved code.

Here is panic https://github.com/kubernetes-sigs/external-dns/blob/2de3b50b14681933a1efea38c4ccf59bed1c0a6f/controller/execute.go#L451

If there is a panic, then there is something else

Reply:
There is a panic because resp is nil and it's called just after: `resp.status`.